lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -2025,10 +2028,14 @@ Attribute DotOperandEncodingAttr::parse(AsmParser &parser, Type type) {
 }
 
 void DotOperandEncodingAttr::print(mlir::AsmPrinter &printer) const {
-  auto mmaParent = getParent().dyn_cast<NvidiaMmaEncodingAttr>();
+  auto parent = getParent();
+  auto mmaParent = parent.dyn_cast<NvidiaMmaEncodingAttr>();
   printer << "<{"
           << "opIdx = " << getOpIdx() << ", parent = " << getParent();
```

Comment:
If we defined `parent` in the above then we can use that here instead of calling `getParent()` again.